ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java

```diff
@@ -164,15 +160,15 @@ private synchronized boolean createRecoveryTaskFromBlock(final SignedBeaconBlock
     makeRoomForNewTracker();
     final BeaconBlockBodyDeneb blockBodyDeneb =
         BeaconBlockBodyDeneb.required(block.getMessage().getBody());
-    recoveryTasks.put(
-        block.getRoot(),
-        new RecoveryTask(
-            block.asHeader(),
-            blockBodyDeneb.getBlobKzgCommitments(),
-            miscHelpersFuluSupplier
-                .get()
-                .computeDataColumnKzgCommitmentsInclusionProof(blockBodyDeneb)));
-    return true;
+    return recoveryTasks.putIfAbsent(
```

Comment:
How often do we expect to already have a value for the SlotAndBlockRoot? If we constantly hit this with an existing mapping, we would be better using `computeIfAbsent`.

This is just to avoid wasting time on `computeDataColumnKzgCommitmentsInclusionProof` when we do not need to.

Reply:
Ok as discussed this won't be an issue in this case.